systems/plants/RigidBodyManipulatorURDF.cpp

```diff
@@ -0,0 +1,513 @@
+    inertia->Attribute("ixx", &I(0, 0));
+    inertia->Attribute("ixy", &I(0, 1));
+    I(1, 0) = I(0, 1);
+    inertia->Attribute("ixz", &I(0, 2));
+    I(2, 0) = I(0, 2);
+    inertia->Attribute("iyy", &I(1, 1));
+    inertia->Attribute("iyz", &I(1, 2));
+    I(2, 1) = I(1, 2);
+    inertia->Attribute("izz", &I(2, 2));
+  }
+
+  auto bodyI = transformSpatialInertia(T, static_cast<Gradient<Isometry3d::MatrixType, Eigen::Dynamic>::type*>(NULL), I);
+  body->I = bodyI.value();
+
+  return true;
+}
+
+bool parseVisual(shared_ptr<RigidBody> body, TiXmlElement* node, RigidBodyManipulator* model)
+{
+  // todo:  consider implementing this, but I don't need it yet
+  return true;
+}
+
+bool parseCollision(int body_index, TiXmlElement* node, RigidBodyManipulator* model)
+{
+  RigidBody::CollisionElement ce;
+  ce.T = Isometry3d::Identity();
+  TiXmlElement* origin = node->FirstChildElement("origin");
+  if (origin)
+    poseAttributesToTransform(origin, ce.T.matrix());
+
+  bool create_collision_element(true);
+  const char* attr;
+  ce.shape = DrakeCollision::Shape::UNKNOWN;
+
+  TiXmlElement* geometry_node = node->FirstChildElement("geometry");
+  if (!geometry_node)
+    return false;
+
+  TiXmlElement* shape_node;
+  if ((shape_node = geometry_node->FirstChildElement("box"))) {
+    ce.shape = DrakeCollision::Shape::BOX;
+    double x = 0, y = 0, z = 0;
+    attr = shape_node->Attribute("size");
+    if (attr) {
+      stringstream s(attr);
+      s >> x >> y >> z;
+    }
+    ce.params.push_back(x);
+    ce.params.push_back(y);
+    ce.params.push_back(z);
+  } else if ((shape_node = geometry_node->FirstChildElement("sphere"))) {
+    ce.shape = DrakeCollision::Shape::SPHERE;
+    double r = 0;
+    attr = shape_node->Attribute("radius");
+    if (attr) {
+      stringstream s(attr);
+      s >> r;
+    }
+    ce.params.push_back(r);
+  } else if ((shape_node = geometry_node->FirstChildElement("cylinder"))) {
+    ce.shape = DrakeCollision::CYLINDER;
+    double r = 0, l = 0;
+    attr = shape_node->Attribute("radius");
+    if (attr) {
+      stringstream s(attr);
+      s >> r;
+    }
+    attr = shape_node->Attribute("length");
+    if (attr) {
+      stringstream s(attr);
+      s >> l;
+    }
+    ce.params.push_back(r);
+    ce.params.push_back(l);
+  } else if ((shape_node = geometry_node->FirstChildElement("capsule"))) {
+    ce.shape = DrakeCollision::CAPSULE;
+    double r = 0, l = 0;
+    attr = shape_node->Attribute("radius");
+    if (attr) {
+      stringstream s(attr);
+      s >> r;
+    }
+    attr = shape_node->Attribute("length");
+    if (attr) {
+      stringstream s(attr);
+      s >> l;
+    }
+    ce.params.push_back(r);
+    ce.params.push_back(l);
+  } else if ((shape_node = geometry_node->FirstChildElement("mesh"))) {
+    ce.shape = DrakeCollision::Shape::MESH;
+    cerr << "Warning: mesh collision elements will be ignored (until I re-implement the logic below sans boost)" << endl;
+    create_collision_element = false;
```

Comment:
So, just to confirm, for now we only get mesh collision elements if we construct the RBM with `constructModelmex`?